interface.h

```diff
@@ -64,150 +67,853 @@
 
 
 // ----------------------------------------------------------------------------
 // inputs
 
+/**
+ * @brief Get the number of hardware CV inputs available
+ * 
+ * @return uint8_t Returns number of hardware CV inputs
+ */
 uint8_t get_cv_input_count(void);
+
+/**
+ * @brief Get the CV value for the provided input index. Teletype is currently
+ *        the only Monome module that has a CV input.
+ * 
+ * @param index Hardware CV input number (first input is index 0)
+ * @return int16_t Returns CV value represented as a signed integer with the
+ *         range of -16,384 (-10v) and 16,383 (+10v).
+ */
 int16_t get_cv(uint8_t index);
 
+/**
+ * @brief Get the number of hardware CV gate inputs available
+ * 
+ * @return uint8_t Returns number of hardware CV gate inputs
+ */
 uint8_t get_gate_input_count(void);
+
+/**
+ * @brief Get the CV gate value for the provided input index.
+ * 
+ * @param index Hardware CV gate input number (first input is index 0)
+ * @return uint8_t Returns 0 when CV gate input is low
+ * @return uint8_t Returns 1 when CV gate input is high
+ */
 uint8_t get_gate(uint8_t index);
 
 
 // ----------------------------------------------------------------------------
 // outputs
 
+/**
+ * @brief Get the number of hardware CV outputs available
+ * 
+ * @return uint8_t Returns number of hardware CV outputs
+ */
 uint8_t get_cv_output_count(void);
+
+/**
+ * @brief Set the value of the indicated CV output
+ * 
+ * @param output index value of the CV output
+ * @param value CV value represented as a signed integer with a range of
+ *        -16,384 (-10v) and 16,383 (+10v)
+ */
 void set_cv(uint8_t output, int16_t value);
 
+/**
+ * @brief Get the number of hardware CV gate outputs available
+ * 
+ * @return uint8_t Returns number of hardware CV gate outputs
+ */
 uint8_t get_gate_output_count(void);
+
+/**
+ * @brief Set the value of the indicated CV gate output
+ * 
+ * @param output index value of the CV gate output
+ * @param on 0 CV gate output is low,
+ *           1 CV gate output is high
+ */
 void set_gate(uint8_t output, uint8_t on);
 
 
 // ----------------------------------------------------------------------------
 // controls
 
+/**
+ * @brief Get the number of hardware buttons available
+ * 
+ * @return uint8_t Returns number of hardware buttons
+ */
 uint8_t get_button_count(void);
+
+/**
+ * @brief Get the button press status for the button at the indicated index
+ * 
+ * @param index Button number (first button is index 0)
+ * @return uint8_t Returns 0 button not pressed
+ * @return uint8_t Returns 1 button pressed
+ */
 uint8_t is_button_pressed(uint8_t index);
 
+/**
+ * @brief Get the number of hardware knobs available
+ * 
+ * @return uint8_t Returns number of hardware knobs
+ */
 uint8_t get_knob_count(void);
+
+/**
+ * @brief Get the knob value for the provided knob index.
+ * 
+ * @param index Hardware knob number.
+ * @return uint16_t Returns a value between 0 and 255
```

Comment:
it should return values in the range of 0..`MAX_LEVEL`, but entirely possible there is a bug or i didn't keep it consistent - is it what you observed?

Reply:
Huh. I'm not sure why I had documented it that way... I'm still getting around to soldering up FTDI for logging. I will report back soon.

Reply:
I can't seem to find my FTDI/USB cable now. Must be at the office. I will document the desired value of `MAX_LEVEL` and follow up in a separate PR if it is not actually `MAX_LEVEL`